modular-framework/module.h

```diff
@@ -0,0 +1,205 @@
+/* Copyright (c) 2017, ARM Limited. All rights reserved.
+ *
+ * Copyright (c) 2017, Linaro Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * @file
+ *
+ * Modular programming framework supports runtime selectable
+ * implementations for variant software subsystems.
+ *
+ * Multiple implementations of the same subsystem can be built
+ * into individual static libraries or loadable DSOs, and use
+ * constructor functions to register themselves.
+ *
+ * A subsystem can choose one active implementation and provide
+ * APIs to switch between implementations.
```

Comment:
Is there an example of calling a subsystem API and how that gets translated into the module?  I am having trouble finding use of function pointers here.